meeko/flexibility.py

```diff
@@ -5,275 +5,269 @@
 #
 
 from copy import deepcopy
-from collections import defaultdict
-from itertools import product
-from operator import itemgetter
 from .utils import pdbutils
 
 
-class FlexibilityBuilder:
-
-    def __call__(self, setup, freeze_bonds=None, root_atom_index=None, break_combo_data=None, bonds_in_rigid_rings=None, glue_pseudo_atoms=None):
-        """ """
-        self.setup = setup
-        self.flexibility_models = {}
-        self._frozen_bonds = []
-        if not freeze_bonds is None:
-            self._frozen_bonds = freeze_bonds[:]
-        # build graph for standard molecule (no open macrocycle rings)
-        model = self.build_rigid_body_connectivity()
-        model = self.set_graph_root(model, root_atom_index) # finds root if root_atom_index==None
-        self.add_flex_model(model, score=False)
-
-        # evaluate possible graphs for various ring openings
-        if break_combo_data is not None:
-            bond_break_combos = break_combo_data['bond_break_combos']
-            bond_break_scores = break_combo_data['bond_break_scores']
-            broken_rings_list = break_combo_data['broken_rings']
-            for index in range(len(bond_break_combos)):
-                bond_break_combo = bond_break_combos[index]
-                bond_break_score = bond_break_scores[index]
-                broken_rings =     broken_rings_list[index]
-                model = self.build_rigid_body_connectivity(bond_break_combo, broken_rings, bonds_in_rigid_rings, glue_pseudo_atoms)
-                self.set_graph_root(model, root_atom_index) # finds root if root_atom_index==None
-                self.add_flex_model(model, score=True, initial_score=bond_break_score)
-
-        self.select_best_model()
-
-        # clean up
-        del self._frozen_bonds
-        del self.flexibility_models
-        return self.setup
-
-
-    def select_best_model(self):
-        """
-        select flexibility model with best complexity score
-        """
-        if len(self.flexibility_models) == 1: # no macrocyle open rings
-            best_model = list(self.flexibility_models.values())[0]
+def _calc_max_weighted_depth(model, seed_node, bonds_to_break, visited=None, depth=0):
```

Comment:
This file probably needs a few function level comments, but I think I can handle some of that to get it started/make it easier

Reply:
added a couple in ac2da0724891385abdebe0a207e7a1fb7a202453